[OpenStack]Add codes to check the count of nodes(members)

FengyunPan opened the pull request:
After merging this PR(#53146), if there is no available nodes for
the loadbalancer service, UpdateLoadBalancer() will run panic.

**Release note**:
```release-note
NONE
```

dims commented on pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:
```diff
@@ -623,6 +623,10 @@ func getSubnetIDForLB(compute *gophercloud.ServiceClient, node v1.Node) (string,
 	glog.V(4).Infof("EnsureLoadBalancer(%v, %v, %v, %v, %v, %v, %v)", clusterName, apiService.Namespace, apiService.Name, apiService.Spec.LoadBalancerIP, apiService.Spec.Ports, nodes, apiService.Annotations)
 
+	if len(nodes) == 0 {
+		return nil, fmt.Errorf("There is no available nodes for LoadBalancer service %s/%s", apiService.Namespace, apiService.Name)
```
"There are" instead of "There is"

FengyunPan replied:
Oops, done.

dims commented on pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:
```diff
@@ -1024,7 +1028,7 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(clusterName string, service *v1.Service
 	glog.V(4).Infof("UpdateLoadBalancer(%v, %v, %v)", clusterName, loadBalancerName, nodes)
 
-	if len(lbaas.opts.SubnetId) == 0 {
+	if len(lbaas.opts.SubnetId) == 0 && len(nodes) != 0 {
```
`len(nodes) > 0`?

FengyunPan replied:
Right, Done.

dims commented on pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:
```diff
@@ -1322,6 +1326,10 @@ func (lb *LbaasV1) GetLoadBalancer(clusterName string, service *v1.Service) (*v1
 	glog.V(4).Infof("EnsureLoadBalancer(%v, %v, %v, %v, %v, %v, %v)", clusterName, apiService.Namespace, apiService.Name, apiService.Spec.LoadBalancerIP, apiService.Spec.Ports, nodes, apiService.Annotations)
 
+	if len(nodes) == 0 {
+		return nil, fmt.Errorf("There is no available nodes for LoadBalancer service %s/%s", apiService.Namespace, apiService.Name)
```
"There are" like above

FengyunPan replied:
Done.

dims reviewed (changes requested):
please see inline comments

xsgordon commented:
/sig openstack

Commit:
[OpenStack]Add codes to check the count of nodes(members)

After merging this PR(#53146), if there is no available nodes for
the loadbalancer service, UpdateLoadBalancer() will run panic.

dims commented:
/approve no-issue
/lgtm
